app/questionnaire/navigator.py

```diff
@@ -266,7 +238,7 @@ def get_latest_location(self, completed_blocks=None):
         """
         location_path = self.get_location_path()
         if completed_blocks:
-            incomplete_blocks = [item for item in location_path if item not in completed_blocks]
+            incomplete_blocks = [item for item in location_path if item.__dict__ not in completed_blocks]
```

Comment:
True, although this is implicitly stating that a location __dict__ matches a block dict structure; i.e. all code using this method of comparison is making that assumption rather than Location object itself doing so. Not sure what's better, leave it your discretion.